packages/web-client/app/components/card-pay/deposit-workflow/index.ts

```diff
@@ -153,11 +158,54 @@ class DepositWorkflow extends Workflow {
 }
 
 class DepositWorkflowComponent extends Component {
+  @service declare layer1Network: Layer1Network;
+  @service declare layer2Network: Layer2Network;
+  @tracked layer1Disconnected = false;
+  @tracked layer2Disconnected = false;
+
   workflow!: DepositWorkflow;
   constructor(owner: unknown, args: {}) {
     super(owner, args);
     this.workflow = new DepositWorkflow(getOwner(this));
   }
+
+  setCancelation() {
+    if (!(this.layer1Disconnected || this.layer2Disconnected))
+      this.workflow.setCancelation(undefined);
+    else {
+      let message = '';
+
+      if (this.layer1Disconnected && this.layer2Disconnected) {
+        message =
+          'It looks like your wallets have been disconnected. If you still want to deposit funds, please start again by connecting your wallets.';
+      } else if (this.layer1Disconnected) {
+        message =
+          'It looks like your mainnet wallet has been disconnected. If you still want to deposit funds, please start again by connecting your wallet.';
+      } else if (this.layer2Disconnected) {
+        message =
+          'It looks like your xDai chain wallet has been disconnected. If you still want to deposit funds, please start again by connecting your wallet.';
+      }
+
+      let result = new PostableCollection([
+        new WorkflowMessage({
+          author: cardbot,
+          message,
+        }),
+      ]);
```

Comment:
Why imperatively add this to the workflow when all the other workflow posts are declared in the definition with conditional logic about whether they should be shown?